lib/data_filter.py

```diff
@@ -5,23 +5,64 @@
 def extract_dates_from_filenames(data_dir):
     dates = []
     for filename in os.listdir(data_dir):
-        if filename.endswith('.txt'):
-            date_part = filename.split('_')[0]  # Assumes filenames are in the format 'YYYY-MM-DD_otherinfo.txt'
-            dates.append(date_part)
+        if filename.endswith('.txt') or filename.endswith('.tx0'):
+            parts = filename.split('_')
+            if len(parts) >= 2:
+                file_type = parts[0]
+                date_part = '_'.join(parts[1:]).split('.')[0]
+
+                if file_type == 'DD':
+                    try:
+                        date = pd.to_datetime(date_part, format='%Y_%m_%d').strftime('%Y-%m-%d')
+                        dates.append(date)
+                    except ValueError:
+                        print(f"Invalid date format in filename: {filename}")
+                elif file_type == 'WP2':
+                    try:
+                        date = pd.to_datetime(date_part, format='%d_%m_%Y').strftime('%Y-%m-%d')
+                        dates.append(date)
+                    except ValueError:
+                        print(f"Invalid date format in filename: {filename}")
+                else:
+                    try:
+                        date = pd.to_datetime(parts[0], format='%Y-%m-%d').strftime('%Y-%m-%d')
+                        dates.append(date)
+                    except ValueError:
+                        print(f"Invalid date format in filename: {filename}")
+            else:
+                print(f"Filename format is not recognized: {filename}")
     return dates
 
 
 def filter_temperature_data(temperature_file, dates, output_file):
-    # Load the original temperature data
-    temp_df = pd.read_csv(temperature_file, sep="\t", header=0, parse_dates=[0], dayfirst=True)
+    try:
+        # Load the original temperature data
+        temp_df = pd.read_csv(temperature_file, sep="\t", header=0, parse_dates=['time'], dayfirst=True)
+        print(f"Temperature data loaded successfully: {temp_df.head()}")
+
+        # Ensure the 'time' column is in datetime format
+        if not pd.api.types.is_datetime64_any_dtype(temp_df['time']):
+            temp_df['time'] = pd.to_datetime(temp_df['time'], errors='coerce')
+            print(f"Time column converted to datetime: {temp_df['time'].head()}")
+
+        # Check if there are any NaT values after conversion
+        if temp_df['time'].isnull().any():
+            print("Warning: Some 'time' values could not be converted to datetime. Check for malformed data.")
+            print(temp_df[temp_df['time'].isnull()])
+
+        # Convert dates to pandas datetime for matching
+        temp_df['date'] = temp_df['time'].dt.strftime('%Y-%m-%d')
+        print(f"Processed date column: {temp_df['date'].unique()}")
```

Comment:
I believe that log files should be used to record information, rather than printing it to the console.